src/VariableCollector/DynamicFilterVariables.php

```diff
@@ -31,13 +35,22 @@ public function __construct(CompilerInterface $compiler, TypeStringResolver  $ty
         $this->closureTypeFactory = $closureTypeFactory;
     }
 
+    /**
+     * @param array<string, string|array{string, string}|array{object, string}|callable> $filters
+     */
+    public function addFilters(array $filters): void
+    {
+        $this->tmpFilters = array_merge($this->tmpFilters, $filters);
```

Comment:
This would leave filters added for single presenter/control for all following no? This kind of state can have unexpected side effects.

Reply:
I think aybe filter variables should not be mixed with normal variables like this with global state but covered separately? Maybe AddVarTypesNode visitor can get list of variables and list of filters and do this logic there? Or maybe separate AddFilterVariablesVisitor?

Reply:
I know this is bad practice. I'm reseting it at the end of collect method. It's temporary. There will be no global variables / filters / components / forms because of standalone templates (like new Engine() for newsletters). We will collect all types in resolvers. I have to create issue for that.